tests/foreman/ui/test_host.py

```diff
@@ -868,6 +868,89 @@ def test_positive_search_by_parameter(session, module_org, smart_proxy_location,
         assert values[0]['Name'] == param_host.name
 
 
+@pytest.mark.tier2
+@pytest.mark.rhel_ver_match('8')
+@pytest.mark.no_containers
+def test_positive_search_by_reported_data(
+    target_sat, rhel_contenthost, module_org, module_ak_with_cv
+):
+    """
+    Search for host by reported data.
+    For example, search by `reported.bios_vendor = SeaBIOS`.
+
+    :id: 54341d00-34bc-11ef-a8a6-000c29a0e355
+
+    :expectedresults: Return only hosts matching the reported data.
+
+    :Verifies: SAT-9132
+
+    :customerscenario: true
+    """
+    result = rhel_contenthost.register(module_org, None, module_ak_with_cv.name, target_sat)
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    client = rhel_contenthost
+
+    reported_data_params = [
+        'bios_release_date',
+        'bios_vendor',
+        'bios_version',
+    ]
+    source_host = target_sat.api.Host().search(query={'search': f'name={client.hostname}'})[0]
+    assert source_host.reported_data, f'Source host {client.hostname} does not report any data.'
+    source_reported_data = source_host.reported_data
+
+    with target_sat.ui_session() as session:
+        session.organization.select(org_name=ANY_CONTEXT['org'])
+        session.location.select(loc_name=ANY_CONTEXT['location'])
+
+        for param_name in reported_data_params:
+            param_value = source_reported_data[param_name]
+            search_string = f'reported.{param_name} = {param_value}'
+            api_hosts = target_sat.api.Host().search(query={'search': search_string})
+            ui_hosts = session.host.search(search_string)
+            assert set([host.name for host in api_hosts]) == set(
+                [host['Name'] for host in ui_hosts]
+            )
+
+
+@pytest.mark.tier2
+@pytest.mark.usefixtures('function_host')
```

Comment:
There is no use of this fixture in test, is that intended, if not, can it be removed?

Reply:
Sorry for being late with the reply @Gauravtalreja1 
Ondrej is right, I need **a** host for the test, but I don't need to explicitly know it's name, because I just compare search results from API with search from UI. And the search results can include also other hosts like the Satellite host.